### PR TITLE
Add pagination for Tours admin page

### PR DESCRIPTION
**Pull request by grigiv**

### CODE REVIEW

File: src/main/java/com/epam/controller/ToursController.java [hunk 1]
```diff
@@ -32,6 +34,26 @@ public ModelAndView getToursList() {
         return toursModel;
     }
 
+    @GetMapping("/listoftours/{pageNum}")
+    public ModelAndView getToursListByPage(@PathVariable Integer pageNum, ModelMap modelMap) {
+        ModelAndView toursModel = new ModelAndView();
```
Comment (ghost) [inline src/main/java/com/epam/controller/ToursController.java:48]: it seems like business logic, please replace it to the service layer

File: src/main/java/com/epam/repository/TourOfferDAO.java [hunk 2]
```diff
@@ -20,7 +20,13 @@ public TourOfferDAO(JdbcTemplate JdbcTemplate) {
     }
 
     public List<TourOffer> getTours(){
-        return JdbcTemplate.query("SELECT * FROM tourOffer", (rs, rowNum) -> buildTour(rs));
+        String sql = "SELECT * from tourOffer ";
```
Comment (ghost) [inline src/main/java/com/epam/repository/TourOfferDAO.java]: Do we really need these changes?

File: src/main/java/com/epam/controller/ToursController.java [hunk 3]
```diff
@@ -32,6 +34,26 @@ public ModelAndView getToursList() {
...
+        Integer from = 0;
+        Integer rowsPerPage = toursOfferService.rowsPerPage;
+        Integer totalRows = toursOfferService.getAmountOfTours();
+        Integer totalPages = (totalRows % rowsPerPage == 0) ? totalRows / rowsPerPage : totalRows / rowsPerPage + 1;
```
Comment (ghost) [inline src/main/java/com/epam/controller/ToursController.java]: It would be great to see unit tests for this logic

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/controller/ToursController.java [hunk 4]
```diff
@@ -32,6 +34,26 @@ public ModelAndView getToursList() {
...
+        if(pageNum == 1) {
+            // do nothing
```
Comment (goponenkoev) [inline src/main/java/com/epam/controller/ToursController.java]: Hmm, it looks a little bit strange. Maybe make condition is negative?

File: src/main/java/com/epam/controller/ToursController.java (hunk 1 above)
Comment (grigiv) [reply src/main/java/com/epam/controller/ToursController.java:48]: Don't understand what I should do here

Comment (grigiv) [reply src/main/java/com/epam/controller/ToursController.java:48]: Moved some logic to service layer

**Comment by ghost** [comment]
seems not fixed.... (logic is still in controller)

**Review by ghost — CHANGES_REQUESTED** [review]
see the comment above

File: src/main/java/com/epam/controller/ToursController.java [hunk 5]
```diff
@@ -37,9 +37,19 @@ public ToursController(TourOfferService toursOfferService, ReservationService re
...
+        return "redirect:/listoftours/1";
+    }
+
+    @GetMapping("/listoftours/{pageNum}")
+    public ModelAndView getToursListByPage(@PathVariable Integer pageNum, ModelMap modelMap, RedirectAttributes redirectAttributes) {
         ModelAndView toursModel = new ModelAndView();
-        toursModel.addObject("listOfTours", toursOfferService.getTours());
+        Integer rowsPerPage = toursOfferService.rowsPerPage;
+        Integer totalRows = toursOfferService.getAmountOfTours();
+        Integer totalPages = (totalRows % rowsPerPage == 0) ? totalRows / rowsPerPage : totalRows / rowsPerPage + 1;
```
Comment (ghost) [inline src/main/java/com/epam/controller/ToursController.java]: why this logic is still here? we can move it to the service layer

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by MaltsevaKsenia — APPROVED** [review]

**Review by Cisfinitum — APPROVED** [review]

**Review by goponenkoev — APPROVED** [review]

**Pull request merged**